src/native/runtime-base/monodroid-dl.hh

```diff
@@ -103,7 +150,7 @@ namespace xamarin::android::internal
 
 	public:
 		[[gnu::flatten]]
-		static void* monodroid_dlopen (const char *name, int flags, char **err, [[maybe_unused]] void *user_data) noexcept
+		static void* monodroid_dlopen (const char *name, int flags, char **err, bool use_aot_cache) noexcept
```

Comment:
I think part of my confusion here is thinking that "use aot cache" is an either/or interpretation: either (only) AOT cache is used, *or* AOT cache is *not* used.  (It's a boolean! Is this not what boolean means?)

While the logic appears to instead be: "when true, check AOT *first*, then fallback to non-AOT cache."

Consequently, a better name than `use_aot_cache` could be *`prefer_aot_cache`* or `check_aot_cache_first`.